### DISCUSSION TITLE
Avalonia ui replacement for Paint event

### DISCUSSION THREAD

**Discussion by AviadBly** [AvaloniaUI/Avalonia#12306] [Q&A] [answered]
Hi. the following is an example in good old winforms I would like to convert to avalonia ui. been trying for a while now.
Thanks.
![image](https://github.com/AvaloniaUI/Avalonia/assets/111220386/f48b0012-9134-47b9-8412-52d24594ca3c)

![image](https://github.com/AvaloniaUI/Avalonia/assets/111220386/3d9c3cbd-3282-454d-9244-f5f1401a4cfe)

I've tried messing with DrawingContext but with no success.

**Comment by sn4k3** [comment, marked as answer]
You need to override the Render method, see a sample here: https://github.com/sn4k3/UVtools/blob/master/UVtools.AvaloniaControls/AdvancedImageBox.axaml.cs#L1096

**Reply by AviadBly** [reply]
Thank you very much, 
used Bing AI to learn from that page and it actually helped me fit that information into the style of my project. :)
